rust-runtime/aws-smithy-runtime-api/src/client/interceptors.rs

```diff
@@ -541,47 +537,12 @@ pub trait Interceptor<TxReq, TxRes>: std::fmt::Debug {
     );
 }
 
-pub type SharedInterceptor<TxReq, TxRes> = Arc<dyn Interceptor<TxReq, TxRes> + Send + Sync>;
-
-#[derive(Debug)]
-struct Inner<TxReq, TxRes> {
-    client_interceptors: Vec<SharedInterceptor<TxReq, TxRes>>,
-    operation_interceptors: Vec<SharedInterceptor<TxReq, TxRes>>,
-}
-
-// The compiler isn't smart enough to realize that TxReq and TxRes don't need to implement `Clone`
-impl<TxReq, TxRes> Clone for Inner<TxReq, TxRes> {
-    fn clone(&self) -> Self {
-        Self {
-            client_interceptors: self.client_interceptors.clone(),
-            operation_interceptors: self.operation_interceptors.clone(),
-        }
-    }
-}
-
-#[derive(Debug)]
-pub struct Interceptors<TxReq, TxRes> {
-    inner: Arc<Mutex<Inner<TxReq, TxRes>>>,
-}
+pub type SharedInterceptor = Arc<dyn Interceptor + Send + Sync>;
```

Comment:
FYI, `SharedInterceptor` will probably be a newtype wrapper around `Arc` in one of my later PRs, instead of a type alias so that the `.interceptor` method on `SdkConfig`, which is a public API, can specify `SharedInterceptor` in its signature.